### DISCUSSION TITLE
How to serve standalone javascript from Next.js application

### DISCUSSION THREAD

**Discussion by hitendra-ap** [vercel/next.js#55658] [Help]
### Summary

My requirement is to serve javascript bundle dynamically (without having versioning) to other application which are not necessarily built in using Next.js framework.

Initially I thought of exposing some function through window object but i couldnt find suitable next configurations for the same. We could use module federation but I want to consume the javascript without having any explicit config on consumer side through a cdn.

I tried:

App1 built in Next.js

```
package.json
package-lock.json
next.config.js
pages
myModule
	|_foo.js
```

Inside next.config.js, it contains
```
...other configs
webpack(config, { isServer }) {
    config.resolve = {
      ...config.resolve,
      fallback: {
        fs: false,
      },
    };

    config.node = {
      global: true,
    };

    config.plugins.push(
      new NextFederationPlugin({
        name: 'MyModule',
        library: { type: 'var', name: 'foo' },
        filename: 'static/MyModule.js',
        exposes: {
          'foo': './myModule/foo.js',
        },
        shared: {},
      })
    );

    if(!isServer){
      config.output.publicPath = 'http://localhost:5000/_next/';
    }

    return config;
 }
```

Inside foo.js
```
export function hello(name) {
    console.log(`hello +++++ ${name}`);
};
```

I want to access this in another application running on another domain, could not be necessarily built in Next.js framework so, wanted to inject javascript using cdn.

```
<script src='http://localhost:5000/_next/static/MyModule.js'></script>
```

I was expecting the window object should contain `foo` but its not the case.

Could anyone suggest, what is wrong here?...or any new approach ?

Thank you very much in advance

### Additional information

When I open any new page of my next.js application running on `http://localhost:5000`, I can see the window object contains the `foo` object but it did not happen the other way when I tried to download the specific chunk using script tag above.

### Example

-

**Comment by tlchatt** [comment]
I've been trying to figure this out as well. Want to embed react code as a remote js script and serve it from my server and over write the existing Dom on a non react website. Could not get it to serve a non html doctype and none of the code in my useeffect hooked logged when embedding the remote url as a script tag. Ended up having to iframe but then it's not easy to interact with the existing Dom.
